src/oneid/oneid-ecs-core/src/main/webui/eslint.config.mjs

```diff
@@ -1,179 +1,53 @@
-        },
-      ],
-
-      radix: 'error',
-      'one-var': ['error', 'never'],
-      'object-shorthand': 'error',
-      'no-var': 'error',
-      'no-param-reassign': 'error',
-      'no-underscore-dangle': 'error',
-      'no-undef-init': 'error',
-      'no-throw-literal': 'error',
-      'no-new-wrappers': 'error',
-      'no-eval': 'error',
-      'no-console': 0,
-      'no-caller': 'error',
-      'no-bitwise': 'error',
       eqeqeq: ['error', 'smart'],
-      'max-classes-per-file': ['error', 1],
-      'guard-for-in': 'error',
-      complexity: 'error',
-      'arrow-body-style': 'error',
-      'import/order': 'error',
-      '@typescript-eslint/no-unused-vars': 'off',
-      '@typescript-eslint/explicit-module-boundary-types': 'off',
-      '@typescript-eslint/no-inferrable-types': 'off',
-      '@typescript-eslint/no-explicit-any': 'off',
-
-      '@typescript-eslint/array-type': [
-        'error',
-        {
-          default: 'generic',
-        },
-      ],
-
-      '@typescript-eslint/await-thenable': 'error',
-      '@typescript-eslint/consistent-type-assertions': 'error',
-      '@typescript-eslint/dot-notation': 'error',
-
-      '@typescript-eslint/member-delimiter-style': [
-        'error',
-        {
-          multiline: {
-            delimiter: 'semi',
-            requireLast: true,
-          },
-
-          singleline: {
-            delimiter: 'semi',
-            requireLast: false,
-          },
-        },
-      ],
-
-      '@typescript-eslint/no-floating-promises': 'error',
-      'no-unused-expressions': 'off',
-      '@typescript-eslint/no-unused-expressions': ['error'],
-      '@typescript-eslint/prefer-function-type': 'error',
-      '@typescript-eslint/restrict-plus-operands': 'error',
-      semi: 'off',
-      '@typescript-eslint/semi': ['error'],
-      '@typescript-eslint/unified-signatures': 'error',
-      'react/prop-types': 'off',
-      'react/display-name': 'off',
-      'react/jsx-key': 'error',
-
-      'react/jsx-no-bind': [
-        'error',
-        {
-          allowArrowFunctions: true,
-        },
-      ],
-
-      'react-hooks/rules-of-hooks': 'warn',
-      'functional/no-let': 'error',
-      'functional/immutable-data': 'warn',
-      'sonarjs/no-small-switch': 'off',
-      'sonarjs/no-duplicate-string': 'off',
-      'sonarjs/no-nested-template-literals': 'warn',
-
-      '@typescript-eslint/no-empty-function': [
-        'error',
-        {
-          allow: ['arrowFunctions'],
-        },
-      ],
-
       'react/jsx-uses-react': 'off',
       'react/react-in-jsx-scope': 'off',
+      'functional/immutable-data': 'error',
+      '@typescript-eslint/consistent-type-definitions': ['warn', 'type'],
+      '@typescript-eslint/array-type': ['warn', { default: 'generic' }],
+    },
+  },
+  {
+    plugins: {
+      'react-hooks': reactHooks,
     },
+    rules: {
+      ...reactHooks.configs.recommended.rules,
+    },
+  },
+  {
+    files: ['*.mjs', '*.cjs', '*.js?(x)'],
```

Comment:
shouldn't we include also `*.ts?(x)` files also

Reply:
This elint configuration load the files included in the tsconfig.json by default, so they should be already icluded